scipy/io/tests/test_mmio.py

```diff
@@ -387,6 +387,20 @@ def test_sparse_formats(self):
                 result = mmread(fn).todense()
                 assert_array_almost_equal(result, expected)
 
+    def test_precision(self):
+        for precision in range(1, 8):
+            # construct sparse matrix with pi as last main diagonal entry
+            n = 10**precision + 1
+            A = scipy.sparse.lil_matrix((n, n))
+            A[n-1, n-1] = pi
+            # write matrix with low precision and read again
+            mmwrite(self.fn, A, precision=precision)
+            A = scipy.io.mmread(self.fn)
+            # check for right entries in matrix
+            assert_array_equal(A.row, [n-1])
+            assert_array_equal(A.col, [n-1])
+            assert_array_almost_equal(A.data, [pi], decimal=precision-1)
```

Comment:
Yes, that would be better.

I am also wondering if the format
`fmt = ('%i', '%i') + ('%%.%dg' % precision,) * (IJV.shape[1]-2)`
is what the user expects. What should precision describe exactly? The documentation is somewhat unclear. If it should be the decimal places, `g` must be replaced by `f`.

Otherwise your test would fail if for example the matrix entry would be `10**(-5)` and `precision = 2`.